### PR TITLE
Feature/player 1350

### PR DESCRIPTION
**Pull request by IvanSakharovskii**

### CODE REVIEW

File: js/skin.js [hunk 1]
```diff
@@ -73,17 +73,40 @@ var Skin = React.createClass({
     });
   },
 
+  /**
+   * @public
+   * @description - returns the correct coordinates of events depending on the platform
+   * @param e - event
+   * @returns {object} - coordinates x, y
+   */
+  getCoords: function(e) {
```
Comment (jpfranco) [inline js/skin.js]: I think this should be moved to `Utils.js`. It's not directly related to `skin.js` logic

File: js/controller.js [hunk 2]
```diff
@@ -1293,10 +1293,16 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       }
       // no support
       else {
-        if(this.state.isFullWindow) {
-          this.exitFullWindow();
+
+        if(OO.isIos && this.videoVr){
+          // toggle fullscreen with bitmovin api
+          this.mb.publish(OO.EVENTS.TOGGLE_FULLSCREEN_VR, this.focusedElement);
```
Comment (jpfranco) [inline js/controller.js]: Can we add more details about why the usual fullscreen mechanism doesn't work for VR?

File: js/controller.js [hunk 3]
```diff
@@ -1293,10 +1293,16 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       }
       // no support
       else {
-        if(this.state.isFullWindow) {
-          this.exitFullWindow();
+
+        if(OO.isIos && this.videoVr){
```
Comment (jpfranco) [inline js/controller.js]: Do we need a unit test for this condition?

File: js/views/pauseScreen.js [hunk 4]
```diff
@@ -53,19 +53,22 @@ var PauseScreen = React.createClass({
...
 
   handlePlayerMouseDown: function(e) {
+    e.persist();
     this.props.controller.state.accessibilityControlsEnabled = true;
     this.props.controller.state.isClickedOutside = false;
     this.props.handleVrPlayerMouseDown(e);
   },
   handlePlayerMouseMove: function(e) {
     e.preventDefault();
+    e.persist();
```
Comment (jpfranco) [inline js/views/pauseScreen.js]: Please make sure that these `preventDefault`'s, etc. are not applied to all videos if they are only needed for VR360. Not sure if this is the case here, but we should check.

**Comment by pilievOoyala** [comment]
Look fine aside from JP's comments.

File: js/skin.js (hunk 1 above)
Comment (yvinokurov) [reply js/skin.js]: fix

File: js/controller.js (hunk 3 above)
Comment (yvinokurov) [reply js/controller.js]: Ok. Fix

File: js/controller.js (hunk 2 above)
Comment (yvinokurov) [reply js/controller.js]: this.state.isVideoFullScreenSupported == false;
this.state.isFullScreenSupported == false; 
The rest also did not work. fullscreen bitmovin works correctly

File: js/components/controlBar.js [hunk 5]
```diff
@@ -101,13 +101,70 @@ var ControlBar = React.createClass({
...
+        this.unlockScreenOrientation();
+      }
+    }
+  },
+
+  /**
+   * @description set landscape orientation if it is possible
+   * @private
+   */
+  setLandscapeScreenOrientation: function() {
```
Comment (pilievOoyala) [inline js/components/controlBar.js]: if this is private can we put a "_" in the front?

File: js/components/controlBar.js [hunk 6]
```diff
@@ -101,13 +101,70 @@ var ControlBar = React.createClass({
...
+      }
     }
-    if(this.props.controller && typeof this.props.controller.toggleStereoVr === "function") {
+  },
+
+  /**
+   * @description set possibility to use all orientations
+   * @private
+   */
+  unlockScreenOrientation: function() {
```
Comment (pilievOoyala) [inline js/components/controlBar.js]: here too

File: js/controller.js (hunk 2 above)
Comment (pilievOoyala) [reply js/controller.js]: Can you add this in as a comment?

**Pull request merged**